### PR TITLE
Fixes for GUI startup sequence

### PR DESCRIPTION
**Pull request by ichorid**
This PR adds some fixes related to #6582 

Also, fixes error buffering broken after startup refactoring.

### CODE REVIEW

**Comment by ichorid** [comment]
retest this please

**Comment by ichorid** [comment]
@drew2a , I removed Sentry log skipping, I guess it was too broad. Was it right?

**Comment by drew2a** [comment]
@ichorid not sure. 

The line that you've removed solved the problem (https://github.com/Tribler/tribler/commit/69125a7ede3c1b4950400e295ada9f987581ac75) of error spam that occurs in the case of (I guess) lost connection with the core (ref: https://github.com/Tribler/tribler/pull/5821).

So from my point of view, this line should not be removed.

File: src/tribler-gui/tribler_gui/event_request_manager.py [hunk 1]
```diff
@@ -71,7 +73,6 @@ def events_start_received(self, event_dict):
     def on_error(self, error, reschedule_on_err):
         self._logger.info(f"Got Tribler core error: {error}")
 
-        SentryReporter.ignore_logger(self._logger.name)
```
Comment (drew2a) [inline src/tribler-gui/tribler_gui/event_request_manager.py:74]: **Issue:** without this line we will have unuseful garbage in the logs, see https://github.com/Tribler/tribler/pull/5821

**Review by drew2a — CHANGES_REQUESTED** [review]

File: src/tribler-gui/tribler_gui/event_request_manager.py (hunk 1 above)
Comment (ichorid) [reply src/tribler-gui/tribler_gui/event_request_manager.py:74]: Addressed by logging only every 10th connection attempt.

**Comment by ichorid** [comment]
retest this please

**Comment by ichorid** [comment]
retest this please

File: src/tribler-core/tribler_core/components/restapi/restapi_component.py [hunk 2]
```diff
@@ -126,6 +126,9 @@ def report_callback(reported_error: ReportedError):
             self._events_endpoint.on_tribler_exception(reported_error)
 
         self._core_exception_handler.report_callback = report_callback
+        # Reraise the unreported error, if there is one
+        if self._core_exception_handler.unreported_error:
+            report_callback(self._core_exception_handler.unreported_error)
```
Comment (drew2a) [inline src/tribler-core/tribler_core/components/restapi/restapi_component.py:131]: **Note** it is better to set `unreported_error` to None to indicate that the error has been reported.

```suggestion
            report_callback(self._core_exception_handler.unreported_error)
            self._core_exception_handler.unreported_error = None
```

File: src/tribler-gui/tribler_gui/event_request_manager.py [hunk 3]
```diff
@@ -69,20 +75,23 @@ def events_start_received(self, event_dict):
                 SentryReporter.set_user(public_key.encode('utf-8'))
 
     def on_error(self, error, reschedule_on_err):
-        self._logger.info(f"Got Tribler core error: {error}")
+        if error == QNetworkReply.ConnectionRefusedError:
+            if self.remaining_connection_attempts % 10 == 0:
+                self._logger.info("Tribler Core refused connection, retrying...")
```
Comment (drew2a) [inline src/tribler-gui/tribler_gui/event_request_manager.py:80]: **Issue** it is unclear, why should we log each 10th attempt?
From my POV  it is better either to remove logging (line 80) or to replace it by:

```suggestion
            self._logger.debug("Tribler Core refused connection, retrying...")
```

File: src/tribler-gui/tribler_gui/event_request_manager.py [hunk 4]
```diff
@@ -118,14 +127,13 @@ def on_finished(self):
...
-        self.connect_timer = QTimer()
-        self.connect_timer.setSingleShot(True)
-        self.connect_timer.timeout.connect(self.connect)
-        self.connect_timer.start(500)
+        self.connect_timer.start(RECONNECT_INTERVAL_MS)
 
     def connect(self, reschedule_on_err=True):
-        self._logger.info("Will connect to events endpoint")
+        if self.remaining_connection_attempts % 10 == 0:
+            self._logger.info("Will connect to events endpoint")
```
Comment (drew2a) [inline src/tribler-gui/tribler_gui/event_request_manager.py:134]: **Issue** same here. It is unclear why should we log each 10th attempt.

**Review by drew2a — CHANGES_REQUESTED** [review]

File: src/tribler-gui/tribler_gui/event_request_manager.py (hunk 3 above)
Comment (ichorid) [reply src/tribler-gui/tribler_gui/event_request_manager.py:80]: If I understood you correctly before, your original concern was about the logger spamming the breadcrumbs list. Well, logging each 10th attempt solves that problem. Seeing the connection attempts at the "Info" level is really useful during development.

 I'm no expert with Sentry internals, but if you think changing the logging level to Debug mode will solve the breadcrumbs spam problem, so be it.

Comment (drew2a) [reply src/tribler-gui/tribler_gui/event_request_manager.py]: Breadcrumbs are configurable. Now we use these settings:
https://github.com/Tribler/tribler/blob/151e7ab936187717786b05736cd0dcf4b455340a/src/tribler-common/tribler_common/sentry_reporter/sentry_reporter.py#L114-L117

Comment (ichorid) [reply src/tribler-gui/tribler_gui/event_request_manager.py]: BTW, wouldn't enabling Sentry to capture Debug-level logs enhance our ability to debug problems? Or that will produce too much output?

**Comment by ichorid** [comment]
retest this please

**Review by drew2a — APPROVED** [review]

File: src/tribler-gui/tribler_gui/event_request_manager.py (hunk 3 above)
Comment (drew2a) [reply src/tribler-gui/tribler_gui/event_request_manager.py]: I'm not sure, but I guess so :)

**Comment by ichorid** [comment]
retest this please

**Pull request merged**